constraint-solver/src/indexed_constraint_system.rs

```diff
@@ -392,8 +441,21 @@ fn variable_occurrences<T: RuntimeConstant, V: Hash + Eq + Clone>(
                 .unique()
                 .map(move |v| (v.clone(), ConstraintSystemItem::BusInteraction(i)))
         });
+    let occurrences_in_derived_variables = constraint_system
+        .derived_variables
+        .iter()
+        .enumerate()
+        // We ignore the derived variables itself because it is not a constraint
+        // and does not matter in substitutions (if we substitute the derived
```

Comment:
Can you explain this? If we have `a := 1/b` we do not register `a` as being accessed in this item. Why?

Reply:
Yes. The occurrence map is used for substitutions: If we substitute `b` by `y` we also want to do that inside the computation method. But if we substitute `a` by `t`, we don't really care, we also don't want to substitute `t := 1/b` because `t` might not be a derived variable (if it is, it has its own computation method). Also if we want to figure out the referenced unused variables, we don't want to count `a` in here.